### ISSUE TITLE
Boolean column with filter collection and render html not working, displays true/false

### ISSUE THREAD

**Issue by irg1008** [ghiscoding/Angular-Slickgrid#1240]
### Describe the bug

When setting a column type of "boolean", and then assigning a collection to display custom values, it works. Then if I add some injected html into the collection label with `labelPrefix`, instead of displaying the label, the value "true" or "false" are displayed as labels

If I disable this the label is correctly displayed, but the html is not injected. Thank you

See the "HERE" note in the next snippet :
```typescript
  GetCheckboxGridColumn(field: string, group: FeedColumnCategory, size = 75, editable = true): Column {
    return {
     ...
      type: FieldType.boolean,
      sortable: true,
      formatter: Formatters.multiple,
      params: { formatters: [Formatters.checkmark, backupFormatter] },
      minWidth: size,
      filterable: true,
      editor: editable ? { model: Editors.checkbox } : null,
      filter: {
        // HERE: Displaying "true" and "false" as labels when "enableRenderHtml" set to true
        enableRenderHtml: true,
        collection: [
          { value: '', label: 'Todos' },
          { value: true, label: 'Verdadero', labelPrefix: `<i class="fa fa-check"></i> ` },
          { value: false, label: 'Falso', labelPrefix: `<i class="fa fa-close"></i> ` },
        ],
        model: Filters.singleSelect,
      },
      ...
    };
  }
```

### Reproduction

* Create a column with boolean type
* Add custom collections for custom labels
* Set `enableRenderHtml` to true
* See how instead of custom labels "true" and "false" are displayed

### Expectation

The html should be injected and displayed as label

### Environment Info

```shell
Angular 16.2.1
Slickgrid 3.2.0
TS 4.9
Chrome 116.0.5845.98
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/angular-slickgrid/wiki/HOWTO---Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
seems to be working fine for me, it is showing as I expect it to show. 

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/643976/3aa602a1-6dd1-49ad-8fad-9e821e674cb3)

**Comment by irg1008** [comment]
I mean in the input, for me it works in the dropdown but the final input content shows "true" or "false"

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/47559427/2e7084a8-0e32-46ca-b729-ae19cb37ea8a)

**Comment by irg1008** [comment]
This is my grid table:
![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/47559427/c94ece9c-2e8b-45d5-b8c0-969990bb33d5)

**Comment by ghiscoding** [comment]
oh ok now I understand what your problem is and I'm able to replicate, it might be something with the new lib I rewrote but the lib itself seems correct as shown in this multiple-select-vanille [demo](https://ghiscoding.github.io/multiple-select-vanilla/#/options27), I'll have to investigate further

**Comment by ghiscoding** [comment]
ohhhhh I see what the problem is, I made a mistake and used the wrong ms-select-vanilla flag, I used `useSelectOptionLabelToHtml` instead of `renderOptionLabelAsHtml` and the first flag is to render html for values instead of labels. I'll push a fix in Slickgrid-Universal but in the meantime you could use a temp fix by providing this
```ts
filter: {
  // ...
  model: Filters.singleSelect,
  filterOptions: { renderOptionLabelAsHtml: true, useSelectOptionLabelToHtml: false }
},
```
and here's the new output with the temp fix until a new version is released sometime this week

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/643976/70392d12-a3db-4fb5-bc7c-79f66b3ee6e1)

**Comment by ghiscoding** [comment]
@irg1008 it should be fixed in latest [v6.2.1](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v6.2.1), I have also updated Slickgrid-Universal [Example 7](https://ghiscoding.github.io/slickgrid-universal/#/example07) to demo the fix (I didn't update any Angular-Slickgrid examples though). Also worth to know that this issue was for both the Filter and the Editor since I use the wrong flag in both

**Comment by irg1008** [comment]
@ghiscoding Amazing, thank you so much for this fix and the library!! Keep up the amazing work
